src/rust/ide/src/test.rs

```diff
@@ -302,3 +368,90 @@ pub fn assert_call_info
         assert_eq!(encountered,&expected_info);
     }
 }
+
+
+
+// ==============
+// === Runner ===
+// ==============
+
+/// Helper that runs given test multiple times simulating different executor interleaving.
+/// The `Runner` can be used to write tests against the race conditions dependent on whether some
+/// code path gets interrupted by executor's run or not.
+///
+/// Typical use-case is to call `Runner::run(|runner| { …test code… })`.
+/// For testing a single, specific iteration `run_with` or `run_nth` should be used.
+///
+/// The test when run is given a handle to the runner object that offers `perhaps_run_until_stalled`
+/// method. It should be invoked (with a fixture object) in places where typically the test code
+/// would use the fixture's `run_until_stalled`. The runner will then either call the
+/// `run_until_stalled` or do nothing.
+///
+/// The `run` method will run test multiple times, to cover all possible combinations.
+/// The test should call `perhaps_run_until_stalled` the same number of times on each iteration.
+/// Otherwise, the `perhaps_run_until_stalled` behavior is unspecified (but still well-formed).
+#[derive(Clone,Copy,Debug,Default)]
+pub struct Runner {
+    /// Incremented each time when the runnee calls an interruption point.
+    /// Reset to 0 after each run.
+    current : u32,
+    /// Bitmap that encodes behavior of subsequent `run_until_stalled` calls. True means running.
+    seed : BitField128,
```

Comment:
Is there any point for keeping 128 bits when the reasonable limit is ~20?

Reply:
should be 32